crates/services/txpool_v2/src/service.rs

```diff
@@ -221,6 +227,18 @@ where
     View: TxPoolPersistentStorage,
 {
     async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
+        // TODO: move this to the Task struct
+        if self.metrics {
+            let pool = self.pool.read();
+            let num_transactions = pool.storage.tx_count();
+
+            let executable_txs =
+                pool.selection_algorithm.number_of_executable_transactions();
+
+            record_number_of_transactions_in_txpool(num_transactions);
+            record_number_of_executable_transactions_in_txpool(executable_txs);
+        }
```

Comment:
It is very bad idea to do it on each iteration of the `run`. The `run` function should be as light as possible.

I think we need to update these values when we insert something into TxPool, or when we remove something from TxPool.

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2385/commits/adba071c42f64be085653258f6c63dc20b61aaa5 and https://github.com/FuelLabs/fuel-core/pull/2385/commits/82262eb819585f57eef77b85b70da8ca009a9f1f

Assumptions:
* Insertion is only done in `pool.insert()` within `fn insert_transaction()` (_+ this may also remove some transactions_)
* All removals are concluded with a call to `update_components_and_caches_on_removal()`